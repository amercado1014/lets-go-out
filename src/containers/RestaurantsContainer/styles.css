.restaurant-container {
  color: white;
}

.location-message {
  background-color: #20161C;
  height: 60vh;
  width: 50vw;
  margin: 2% auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}