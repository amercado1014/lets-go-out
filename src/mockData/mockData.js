export const mockRestaurantsApiData = {
  "address": {
    "apiKey": null,
    "streetAddress": "",
    "latitude": 39.6682515,
    "longitude": -105.096661,
    "city": "Denver",
    "state": "CO",
    "zip": "80227",
    "aptNumber": null
  },
  "restaurants": [
    {
      "apiKey": "aa1163f8eb68a8f7144247b91fbcb2f31932e981e6050b6c",
      "deliveryMin": 0,
      "deliveryPrice": 3.5,
      "logoUrl": "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-s-wadsworth-21136_1407870747356.png",
      "name": "Big Daddy's Pizza - S. Wadsworth",
      "streetAddress": "2772 S Wadsworth Blvd",
      "city": "Denver",
      "state": "CO",
      "zip": "80227",
      "foodTypes": [
        "Pizza"
      ],
      "phone": "(303) 988-3697",
      "latitude": 39.666141,
      "longitude": -105.080677,
      "minFreeDelivery": 0,
      "taxRate": 0.08,
      "acceptsCash": true,
      "acceptsCard": true,
      "offersPickup": true,
      "offersDelivery": true,
      "isTestRestaurant": false,
      "minWaitTime": 45,
      "maxWaitTime": 60,
      "open": false,
      "url": "https://eatstreet.com/denver-co/restaurants/big-daddys-pizza-s-wadsworth",
      "hours": {
        "Monday": [
          "11:00 AM - 3:00 AM"
        ],
        "Saturday": [
          "11:00 AM - 3:00 AM"
        ],
        "Sunday": [
          "11:00 AM - 3:00 AM"
        ],
        "Wednesday": [
          "11:00 AM - 3:00 AM"
        ],
        "Tuesday": [
          "11:00 AM - 3:00 AM"
        ],
        "Friday": [
          "11:00 AM - 3:00 AM"
        ],
        "Thursday": [
          "11:00 AM - 3:00 AM"
        ]
      },
      "timezone": "US/Mountain"
    }
  ]
};

export const returnedMockRestaurantApiData = [
  {
    "apiKey": "aa1163f8eb68a8f7144247b91fbcb2f31932e981e6050b6c",
    "deliveryMin": 0,
    "deliveryPrice": 3.5,
    "logoUrl": "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-s-wadsworth-21136_1407870747356.png",
    "name": "Big Daddy's Pizza - S. Wadsworth",
    "streetAddress": "2772 S Wadsworth Blvd",
    "city": "Denver",
    "state": "CO",
    "zip": "80227",
    "foodTypes": [
      "Pizza"
    ],
    "phone": "(303) 988-3697",
    "latitude": 39.666141,
    "longitude": -105.080677,
    "minFreeDelivery": 0,
    "taxRate": 0.08,
    "acceptsCash": true,
    "acceptsCard": true,
    "offersPickup": true,
    "offersDelivery": true,
    "isTestRestaurant": false,
    "minWaitTime": 45,
    "maxWaitTime": 60,
    "open": false,
    "url": "https://eatstreet.com/denver-co/restaurants/big-daddys-pizza-s-wadsworth",
    "hours": {
      "Monday": [
        "11:00 AM - 3:00 AM"
      ],
      "Saturday": [
        "11:00 AM - 3:00 AM"
      ],
      "Sunday": [
        "11:00 AM - 3:00 AM"
      ],
      "Wednesday": [
        "11:00 AM - 3:00 AM"
      ],
      "Tuesday": [
        "11:00 AM - 3:00 AM"
      ],
      "Friday": [
        "11:00 AM - 3:00 AM"
      ],
      "Thursday": [
        "11:00 AM - 3:00 AM"
      ]
    },
    "timezone": "US/Mountain"
  }
];

export const cleanRestaurantData = {
  apiKey: "aa1163f8eb68a8f7144247b91fbcb2f39e65e39821ec08a9",
  logoUrl: "https://eatstreet-static.s3.amazonaws.com/assets/images/restaurant_logos/big-daddys-pizza-lakewood-21135_1407868711096.png",
  name: "Big Daddy's Pizza - Lakewood",
  streetAddress: "850 Wadsworth Blvd",
  city: "Lakewood",
  state: "CO",
  zip: "80214",
  foodTypes: "Pizza",
  phone: "(303) 736-6090",
  url: "https://eatstreet.com/lakewood-co/restaurants/big-daddys-pizza-lakewood",
  hours: {
    Monday: "11:00 AM - 3:00 AM",
    Saturday: "11:00 AM - 3:00 AM",
    Sunday: "11:00 AM - 3:00 AM",
    Wednesday: "11:00 AM - 3:00 AM",
    Tuesday: "11:00 AM - 3:00 AM",
    Friday: "11:00 AM - 3:00 AM",
    Thursday: "11:00 AM - 3:00 AM"
  }
};