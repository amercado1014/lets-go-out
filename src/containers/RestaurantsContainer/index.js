import React from 'react';
import './styles.css';
import Restaurants from '../Restaurants';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

export const RestaurantsContainer = ({restaurants, locationStatus, 
  favorites}) => {
  const displayRestaurants = restaurants.map(restaurant => {
    let isFavorite = false;

    favorites.forEach(favorite => {
      if (favorite.apiKey === restaurant.apiKey) {
        isFavorite = true;
      }
    });

    return <Restaurants 
      key={restaurant.name} 
      restaurant={restaurant} 
      isFavorite={isFavorite} />;
  });
 
  return (
    <div className="restaurant-container">
      {locationStatus &&
      <div className="location-message">
        <h3>Seems like your browsers GeoLocation is disabled. 
        Please enter a city or zip code to search for restaurants.</h3>
      </div>
      }
      {displayRestaurants}
    </div>
  );
};

export const mapStateToProps = state => ({
  restaurants: state.restaurants,
  locationStatus: state.locationStatus,
  favorites: state.favorites
});

RestaurantsContainer.propTypes = {
  restaurants: PropTypes.array,
  locationStatus: PropTypes.bool,
  favorites: PropTypes.array
};

export default connect(mapStateToProps)(RestaurantsContainer);
