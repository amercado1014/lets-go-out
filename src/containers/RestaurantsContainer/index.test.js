import React from "react";
import { RestaurantsContainer } from "../RestaurantsContainer";
import { shallow } from "enzyme";
import { mockRestaurants } from "../../mockData/mockData";

describe("RestaurantsContainer", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<RestaurantsContainer restaurants={mockRestaurants}/>);
  });

  it("should math the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
