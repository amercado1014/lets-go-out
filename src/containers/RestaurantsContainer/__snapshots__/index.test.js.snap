// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestaurantsContainer should math the snapshot 1`] = `undefined`;
